DarwinWorld/src/main/java/agh/ics/oop/model/mapElements/MapField.java

```diff
@@ -1,30 +1,61 @@
 package agh.ics.oop.model.mapElements;
 
 import agh.ics.oop.model.utils.AnimalComparator;
+import agh.ics.oop.model.utils.Vector2d;
 
+import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.TreeSet;
 
 public class MapField {
-    private TreeSet<Animal> animalsOnField = new TreeSet<>(new AnimalComparator());
+    private LinkedList<Animal> animalsOnField = new LinkedList();
     private boolean isPreferred;
+    private final Vector2d fieldPosition;
     private Plant plant;
 
-    public void addAnimal(Animal animal){
+    public MapField(Vector2d pos) {
+        this.fieldPosition = pos;
+    }
+
+    public void addAnimal(Animal animal) {
         animalsOnField.add(animal);
     }
-    public void removeAnimal(Animal animal) {animalsOnField.remove(animal);}
 
+    public void removeAnimal(Animal animal) {
+        animalsOnField.remove(animal);
+    }
 
     public List<Animal> getAnimalsOnField() {
-        return new LinkedList<>(animalsOnField);
+        return Collections.unmodifiableList(animalsOnField);
     }
 
     public void consumePlant() {
 
     }
-    public void reproduceAnimals() {
-        //możemy zwracać listę powstałych zwierzątek zamiast void
+
+    public Animal reproduceAnimals() {
+        if(animalsOnField.size() >= 2) {
+            List<Animal> parents = findParents();
+            Animal children = new Animal(this.fieldPosition, parents.get(0), parents.get(1));
+            animalsOnField.add(children);
+            return children;
+        }
+
+        return null;
+    }
+
+    private List<Animal> findParents() {
```

Comment:
List<Animal> twoStrongestAnimals = animalsOnField.stream()
                .sorted(AnimalComparator.reversed())
                .limit(2)
                .collect(Collectors.toList()); 
    return twoStrongestAnimals;

Reply:
Zrobiłem tak, żeby liniowo znaleźć rodziców, ale jeżeli nam nie przeszkadza sortowanie całej listy to tak faktycznie jest bardziej przejrzyście.